homeassistant/components/bryant_evolution/climate.py

```diff
@@ -96,130 +98,66 @@ def __init__(
-        if not mode_and_active:
-            raise HomeAssistantError(
-                translation_domain=DOMAIN, translation_key="failed_to_read_hvac_mode"
-            )
-        mode = mode_and_active[0]
-        mode_enum = {
-            "HEAT": HVACMode.HEAT,
-            "COOL": HVACMode.COOL,
-            "AUTO": HVACMode.HEAT_COOL,
-            "OFF": HVACMode.OFF,
-        }.get(mode.upper())
-        if mode_enum is None:
-            raise HomeAssistantError(
-                translation_domain=DOMAIN,
-                translation_key="failed_to_parse_hvac_mode",
-                translation_placeholders={"mode": mode},
-            )
-        return mode_enum
+    def _set_attrs_from_coordinator(self) -> None:
+        # Propagate some parameters that are really system-level, not zone-level,
+        # but that the climate entity needs.
+        self._attr_fan_mode = self.coordinator.read_fan_mode(self.system_id)
+        self._attr_hvac_mode = self.coordinator.read_hvac_mode(self.system_id)
 
-    async def _read_hvac_action(self) -> HVACAction:
-        """Return the current running hvac operation."""
-        mode_and_active = await self._client.read_hvac_mode()
-        if not mode_and_active:
-            raise HomeAssistantError(
-                translation_domain=DOMAIN, translation_key="failed_to_read_hvac_action"
-            )
-        mode, is_active = mode_and_active
-        if not is_active:
-            return HVACAction.OFF
-        match mode.upper():
-            case "HEAT":
-                return HVACAction.HEATING
-            case "COOL":
-                return HVACAction.COOLING
-            case "OFF":
-                return HVACAction.OFF
-            case "AUTO":
-                # In AUTO, we need to figure out what the actual action is
-                # based on the setpoints.
-                if (
-                    self.current_temperature is not None
-                    and self.target_temperature_low is not None
-                ):
-                    if self.current_temperature > self.target_temperature_low:
-                        # If the system is on and the current temperature is
-                        # higher than the point at which heating would activate,
-                        # then we must be cooling.
-                        return HVACAction.COOLING
-                    return HVACAction.HEATING
-        raise HomeAssistantError(
-            translation_domain=DOMAIN,
-            translation_key="failed_to_parse_hvac_mode",
-            translation_placeholders={
-                "mode_and_active": mode_and_active,
-                "current_temperature": str(self.current_temperature),
-                "target_temperature_low": str(self.target_temperature_low),
-            },
+        # Read the zone-level parameters.
+        self._attr_current_temperature = self.coordinator.read_current_temperature(
+            self.system_id, self.zone_id
+        )
```

Comment:
```suggestion
        self._attr_current_temperature = self.zone.curr_temp
```

Reply:
see comment in other reply

Reply:
With respect to this comment:

"Define self.system = coordinator.data.systems[sys_id] and maybe also self.zone = coordinator.data.systems[sys_id].zones[zone_id] in the init for better readability"

I don't think that will work. The coordinator uses frozen dataclasses (immutable objects) that get replaced on every update. Wouldn't this be keeping a reference to the first version of the data only, or do I misunderstand python semantics?

Reply:
The attributes and characteristics of the object are immutable. But you can also do this at the beginning of the set_attr method.

Reply:
sure, but in that case it's not going to escape the function -- is it a big deal in your point of view? I'd rather not expose those maps, and if this just a difference of taste...